[Go] Error building go driver manager (no C++ standard set)

In verifying the release candidate I got


```
# github.com/apache/arrow-adbc/go/adbc/drivermgr
adbc_driver_manager.cc:224:26: warning: default member initializer for non-static data member is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:225:32: warning: default member initializer for non-static data member is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:316:14: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:316:27: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:322:12: error: no member named 'ignore' in namespace 'std'
adbc_driver_manager.cc:349:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:428:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:432:14: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:432:27: warning: range-based for loop is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:473:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:558:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:576:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:648:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:784:3: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
adbc_driver_manager.cc:790:5: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
```

I'm on MacOS M1 (no conda) and have run into this before...there must be at least some compiler default which is very picky about setting the C++ standard.

Maybe https://github.com/apache/arrow-adbc/blob/main/go/adbc/drivermgr/wrapper.go#L22 could gain a `-std=gnu++11` or `-std=c++11`?

--- a/go/adbc/drivermgr/wrapper.go
+++ b/go/adbc/drivermgr/wrapper.go
@@ -20,6 +20,7 @@ package drivermgr
 // #cgo !windows LDFLAGS: -ldl
 // #cgo windows CFLAGS: -DADBC_EXPORTING
 // #cgo windows CPPFLAGS: -DADBC_EXPORTING
+// #cgo CXXFLAGS: -std=c++11
 // #define ADBC_EXPORTING
 // #include "adbc.h"
 // #include <stdlib.h>

